cpp/tests/io/fst/quote_normalization_test.cu

```diff
@@ -0,0 +1,328 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <io/fst/lookup_tables.cuh>
+#include <io/utilities/hostdevice_vector.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/strings/repeat_strings.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream.hpp>
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_buffer.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+namespace {
+
+// Type used to represent the atomic symbol type used within the finite-state machine
+using SymbolT = char;
+using StateT  = char;
+// Type sufficiently large to index symbols within the input and output (may be unsigned)
+using SymbolOffsetT = uint32_t;
+enum class dfa_states : char { TT_OOS = 0U, TT_DQS, TT_SQS, TT_DEC, TT_SEC, TT_NUM_STATES };
+enum class dfa_symbol_group_id : uint32_t {
+  DOUBLE_QUOTE_CHAR,  ///< Quote character SG: "
+  SINGLE_QUOTE_CHAR,  ///< Quote character SG: '
+  ESCAPE_CHAR,        ///< Escape character SG: '\'
+  NEWLINE_CHAR,       ///< Newline character SG: '\n'
+  OTHER_SYMBOLS,      ///< SG implicitly matching all other characters
+  NUM_SYMBOL_GROUPS   ///< Total number of symbol groups
+};
+// Aliases for readability of the transition table
+constexpr auto TT_OOS            = dfa_states::TT_OOS;
+constexpr auto TT_DQS            = dfa_states::TT_DQS;
+constexpr auto TT_SQS            = dfa_states::TT_SQS;
+constexpr auto TT_DEC            = dfa_states::TT_DEC;
+constexpr auto TT_SEC            = dfa_states::TT_SEC;
+constexpr auto TT_NUM_STATES     = static_cast<char>(dfa_states::TT_NUM_STATES);
+constexpr auto NUM_SYMBOL_GROUPS = static_cast<uint32_t>(dfa_symbol_group_id::NUM_SYMBOL_GROUPS);
+// The i-th string representing all the characters of a symbol group
+// TODO: We should reinstantiate this lookup table approach once
+// https://github.com/rapidsai/cudf/pull/14561 is merged std::array<std::vector<SymbolT>,
+// NUM_SYMBOL_GROUPS - 1> const qna_sgs{
+//  {{'\"'}, {'\''}, {'\\'}, {'\n'}}};
+// Temporary workaround:
+struct SymbolToSymbolGroup {
+  CUDF_HOST_DEVICE uint32_t operator()(SymbolT symbol) const
+  {
+    switch (symbol) {
+      case '\"': return static_cast<uint32_t>(dfa_symbol_group_id::DOUBLE_QUOTE_CHAR);
+      case '\'': return static_cast<uint32_t>(dfa_symbol_group_id::SINGLE_QUOTE_CHAR);
+      case '\\': return static_cast<uint32_t>(dfa_symbol_group_id::ESCAPE_CHAR);
+      case '\n': return static_cast<uint32_t>(dfa_symbol_group_id::NEWLINE_CHAR);
+      default: return static_cast<uint32_t>(dfa_symbol_group_id::OTHER_SYMBOLS);
+    }
+  };
+};
```

Comment:
https://github.com/rapidsai/cudf/pull/14561 was merged, so we should be good to revert the workaround.
```suggestion
// The i-th string representing all the characters of a symbol group
std::array<std::vector<SymbolT>, NUM_SYMBOL_GROUPS - 1> const qna_sgs{{{'\"'}, {'\''}, {'\\'}, {'\n'}}};
```